### PR TITLE
Improve virtualbuildenv prepending values for already defined env vars

### LINKED ISSUE #3038: Virtualbuildenv generator should append Visual Studio dirs to current PATH, not completely replace it
I would love to see `set PATH=<Visual Studio dirs>;%PATH%` instead of `set PATH=<generated paths>` in `activate_build.bat` to make possible `activate*.bat` files chaining and preserving current environment values. Virtualbuildenv generator should record difference in OS environment before and after invocation of `vcvarsall.bat` and put only that difference to `activate_build.bat`

### PR DESCRIPTION
**Pull request by danimtb**
- [x] Refer to the issue that supports this Pull Request: closes #3038 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

### CODE REVIEW

File: conans/client/tools/win.py [hunk 1]
```diff
@@ -316,7 +314,15 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
             continue
         try:
             name_var, value = line.split("=", 1)
-            if not only_diff or os.environ.get(name_var) != value:
+            old_value = os.environ.get(name_var)
+            if only_diff:
+                if old_value != value:
+                    if old_value is not None and old_value in value:
+                        new_value = value.replace(os.environ.get(name_var), "")
```
Comment (memsharded) [inline conans/client/tools/win.py]: ``os.environ.get(name_var)`` is ``old_value``

File: conans/client/tools/win.py [hunk 2]
```diff
@@ -316,7 +314,15 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
             continue
         try:
             name_var, value = line.split("=", 1)
-            if not only_diff or os.environ.get(name_var) != value:
+            old_value = os.environ.get(name_var)
```
Comment (memsharded) [inline conans/client/tools/win.py]: ``old_value`` define it only inside the if: minimum possible scope

File: conans/client/tools/win.py [hunk 3]
```diff
@@ -316,7 +314,15 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
...
+            if only_diff:
+                if old_value != value:
+                    if old_value is not None and old_value in value:
```
Comment (memsharded) [inline conans/client/tools/win.py]: No need to check if ``old_value`` is in ``value``, just do the replace directly

Comment (danimtb) [reply conans/client/tools/win.py]: you have to check if old value is in new value in order to append `%<VAR_NAME>%` at the end or not

File: conans/client/tools/win.py [hunk 4]
```diff
@@ -316,7 +314,15 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
             continue
         try:
             name_var, value = line.split("=", 1)
-            if not only_diff or os.environ.get(name_var) != value:
+            if only_diff:
+                old_value = os.environ.get(name_var)
+                if old_value != value:
+                    if old_value is not None and old_value in value:
+                        new_value = value.replace(old_value, "")
+                        new_env[name_var] = new_value + "%{0}%".format(name_var)
```
Comment (lasote) [inline conans/client/tools/win.py]: It won't be expanded in python. So any build helper or recipe using vcvars will get a wrong value here.

File: conans/client/generators/virtualbuildenv.py [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import os
```
Comment (memsharded) [inline conans/client/generators/virtualbuildenv.py]: Unused import?

File: conans/test/util/tools_test.py [hunk 6]
```diff
@@ -747,11 +747,12 @@ def subprocess_check_output_mock(cmd, shell):
         with mock.patch('conans.client.tools.win.vcvars_command', new=vcvars_command_mock):
             with mock.patch('subprocess.check_output', new=subprocess_check_output_mock):
                 vars = tools.vcvars_dict(None, only_diff=False)
-                self.assertEqual(vars["PROCESSOR_ARCHITECTURE"], "AMD64")
-                self.assertEqual(vars["PROCESSOR_IDENTIFIER"], "Intel64 Family 6 Model 158 Stepping 9, GenuineIntel")
-                self.assertEqual(vars["PROCESSOR_LEVEL"], "6")
-                self.assertEqual(vars["PROCESSOR_REVISION"], "9e09")
-                self.assertEqual(vars["ProgramFiles(x86)"], "C:\Program Files (x86)")
+                self.assertEqual("AMD64", vars["PROCESSOR_ARCHITECTURE"])
```
Comment (memsharded) [inline conans/test/util/tools_test.py]: Why this change?

Comment (danimtb) [reply conans/test/util/tools_test.py]: nosetests outputing "expected" value as the "actual" value and "actual" value as "expected"

File: conans/client/generators/virtualenv.py [hunk 7]
```diff
@@ -45,27 +45,31 @@ def format_values(self, flavor, variables):
...
                     # Unquoted variables joined with pathset may look like: PATH=one path;two paths
                     value = ["\"%s\"" % v for v in value] if quote_elements else value
-                    value = path_sep.join(value+[placeholder])
-            else:
-                # single value
+                    value = path_sep.join(value + [placeholder])
+            else:  # single value
+                if old_value != value and old_value is not "" and old_value in value:
+                    # Add placeholder instead of old value
+                    new_value = value.replace(old_value, "")
```
Comment (lasote) [inline conans/client/generators/virtualenv.py]: The old value should be replaced with the placeholder. In the same place. Otherwise, the variable could mess with the order or even break the value consistency.
